### DISCUSSION TITLE
Can Copilot convert/transpile 1000 lines of code between the languages in it's language model?

### DISCUSSION THREAD

**Discussion by TDClarke** [community/community#69060] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I have 1000 lines of PHP that needs converting to java

and 800 lines of Python 2.6 code that i want to convert to Python 3.x,

What is the line/character input and output for copilot?

**Comment by rnldvrgl** [comment]
Transpiling or converting 1000 lines of code between programming languages typically involves a more specialized tool or process. Depending on the complexity of the code and the languages involved, you may need to use dedicated transpilers or migration tools that are specifically designed for such tasks.

**Reply by rnldvrgl** [reply]
Or you can do manual refactoring

**Reply by TDClarke** [reply]
I just want to know the line and character limits before I pay for a month, as bing or chat gpt/gpt4 can do it, it the way they are configured they just can't convert that much code.

**Reply by rnldvrgl** [reply, marked as answer]
Copilot is limited to around 1000 characters in the response. This is mostly a limitation of the OpenAI API.

**Comment by Abkeyev99** [comment]
Thanks so good :)

**Comment by Abkeyev99** [comment]
Click "Marked as answer" Thanks so good :)
